### PR TITLE
Security Updates

### PR DESCRIPTION
**Pull request by dllopis-sfdc**
## What does this PR do?
* Updates upstream packages.
* Supports both Python 2.7 & 3.6.
* Replaces requirements.txt with Pipfile[.lock].

## Why is this change being made?
GitHub [security alerts for requests & Jinja2](https://github.com/krux/krux-marathon-api/network/alerts).

## How does this PR do it?
The code was already compatible with both Python 2 & 3, so that part is a no-op.

## How was this tested? How can the reviewer verify your testing?
<details>
 <summary>Tested locally:…</summary>

```
$ read -s MARATHON_PWD
$ pipenv run python -m krux_marathon_api.cli --user marathon --password "$MARATHON_PWD" --host localhost --port 5080 --log-level info --list-apps
2019-05-30 14:42:55,691: KruxMarathonClient/INFO     : Testing connection to marathon server
host = localhost
port = 5080
2019-05-30 14:42:55,691: marathon_cli/INFO     : Connection success
2019-05-30 14:42:55,692: KruxMarathonClient/INFO     : Listing apps running on marathon
2019-05-30 14:42:56,059: marathon/INFO     : Got response from http://localhost:5080
2019-05-30 14:42:56,067: marathon/WARNING  : Deprecated: Using command as dict instead of string is deprecated
/autoscale-pdx => /usr/local/bin/krux-handle-autoscale --queue PDXAutoscaleSQS --boto-region us-west-2 --interval 60 --log-level info
/krux-consent-account-loader => /usr/local/krux-consent-account-loader/bin/krux-consent-account-loader -Dpidfile.path=/dev/null -Dconfig.file=/usr/local/etc/krux/krux-consent-account-loader.conf -Djava.io.tmpdir=/mnt/tmp
/chronos => java -cp /usr/local/chronos/chronos-3.0.4.jar -Djava.library.path=/usr/local/lib:/usr/lib64:/usr/lib -Xmx512m org.apache.mesos.chronos.scheduler.Main --user c_chronos --mesos_framework_name chronos --zk_hosts zk://zookeeper-mesos-a001.krxd.net:2181,zookeeper-mesos-a002.krxd.net:2181,zookeeper-mesos-a003.krxd.net:2181/ --zk_path /chronos-a --master zk://zookeeper-mesos-a001.krxd.net:2181,zookeeper-mesos-a002.krxd.net:2181,zookeeper-mesos-a003.krxd.net:2181/mesos-a --http_port 5081 --mesos_authentication_principal chronos --mesos_authentication_secret_file /etc/mesos/chronos-credential --mesos_role internal --http_credentials 'chronos:3j6WV4yGxZpBFiFQ3zcs6hB7sUcZbb' --http_realm internal --graphite_host_port kafka-infra001-ash.krxd.net:9518 --graphite_group_prefix stats.counters.prod.mesos-a.chronos --mail_server localhost:25 --mail_from chronos@krxd.net
/autoscale-dub => /usr/local/bin/krux-handle-autoscale --queue DUBAutoscaleSQS --boto-region eu-west-1 --interval 60 --log-level info
/percona-heartbeat => /usr/bin/pt-heartbeat --update --database=percona --host=console.rds-master.krxd.net --defaults-file=/usr/local/etc/krux/console.rds-master.krxd.net-rb_krux.cnf
/percona-deadlock-logger => /usr/bin/pt-deadlock-logger --database=rb_krux --host=console.rds-master.krxd.net --defaults-file=/usr/local/etc/krux/console.rds-master.krxd.net-rb_krux.cnf --dest=D=percona,t=deadlocks
/autoscale-ash => /usr/local/bin/krux-handle-autoscale --queue ASHAutoscaleSQS --boto-region us-east-1 --interval 60 --log-level info
/java-adm-notifications => java -cp /usr/local/java-adm-notifications/java-adm-notifications.jar com.krux.jobs.ADMEmailSender --env prod --log-level INFO --stats --stats-environment prod
/emulator-csv-aggregator-prod => java -classpath /usr/local/krux-har-processor/krux-har-processor.jar com.krux.csv.CsvWriter --env prod --stats --stats-environment prod --log-level info --http-port $PORT0
/data-processing/async-jobs-loader => /usr/local/krux-data-processing-pipeline/bin/async_jobs_loader
```

</details>

## What gif best describes this PR or how it makes you feel?
giphy

## Completion checklist
- [x] The pull request has been appropriately labeled according to
  our [conventions](https://kruxdigital.jira.com/wiki/display/EN/Changes+and+Peer+Review).
- [x] The version of the package has been updated according to [Semantic Versioning](http://semver.org/).
- [x] The change has unit & integration tests as appropriate.
- [x] Documentation is up to date and correct.
- [x] Dependencies are correctly listed under `requirements.pip` and
      are up to date without going over a major version.
- [x] Stakeholders have been notified. Workflow-impacting changes have
      been appropriately socialized to avoid surprises.
- [x] The change is either small or have been canaried in the appropriate environment(s).

### CODE REVIEW

**Comment by dllopis-sfdc** [comment]
I think the build broke because it’s using Python 2.

File: setup.py [hunk 1]
```diff
@@ -31,12 +31,12 @@
     download_url=DOWNLOAD_URL,
     license='All Rights Reserved.',
     packages=find_packages(),
-    # dependencies are named in requirements.pip
-    install_requires=[],
+    install_requires=['krux-stdlib', 'marathon'],
```
Comment (hanpeter) [inline setup.py]: This is an application. Don't do this.

File: setup.py [hunk 2]
```diff
@@ -31,12 +31,12 @@
...
     entry_points={
         'console_scripts': [
             'marathon-api = krux_marathon_api.cli:main',
         ],
     },
     zip_safe=False,  # Don't install a single .egg file, it's harder to debug
+    python_requires='~=2.7, ~=3.6',
```
Comment (hanpeter) [inline setup.py]: Does this actually work on Python 3? Have we tested that?

Comment (dllopis-sfdc) [reply setup.py]: Yes, the test I ran above in the PR I did twice, once in 2.7.16 and once in 3.6.7.

**Pull request merged**